bcda/cclf/cclf.go

```diff
@@ -352,6 +357,71 @@ func getCCLFFileMetadata(filePath string) (cclfFileMetadata, error) {
 	return metadata, nil
 }
 
+
+func getCCLFFileMetadataBCD(filePath string) (cclfFileMetadata, error) {
```

Comment:
Is it possible to combine this logic with the existing function? Maybe the regex can cover both, with variables for which position each field will be in within the regex matches. Or keep the existing check on the name containing `BCD` and use that to set the regex pattern?

Reply:
oh okay, are u saying like have the BCD check and then set the pattern string. And then have an if statement for cclfNumIndx..perfYearIdx etc to be used for the regex matches?

Reply:
Yep! Another option might be named groups in a regex pattern that matches the old and new filenames (instead of using indices). Probably lots of ways to do it :)

Reply:
interesting! I think I was leaning towards separate methods because of the actual logic change for getting the acoID. Its not a simple "filenameMatches[2]" like it used to be for the new format. And also was thinking that this is temporary since it'll just be one way in the future (whenever they decide to phase out the old way). This logic will just be the entire getCCLFFileMetadata. Also wanted to avoid a bunch of if statements lol. Let me think..

Reply:
I thought about this a little more last night (very little haha), But are we getting any technical advantage by making it all into one method? Either by using the named groups regex pattern thing or the indices? Im thinking that this is easier now and for the future when we stop supporting the old legacy way. Then it's just a copy and paste job from one method to the new one. No need to refactor anything. What do u think?

Reply:
That makes sense to me. 👍 Can always combine them later if it turns out we have to support both formats indefinitely.